sdks/java/harness/src/main/java/org/apache/beam/fn/harness/data/PCollectionConsumerRegistry.java

```diff
@@ -352,7 +350,8 @@ public void accept(WindowedValue<T> input) throws Exception {
       // Use the ExecutionStateTracker and enter an appropriate state to track the
       // Process Bundle Execution time metric and also ensure user counters can get an appropriate
       // metrics container.
-      for (ConsumerAndMetadata consumerAndMetadata : consumerAndMetadatas) {
+      for (int size = consumerAndMetadatas.size(), i = 0; i < size; ++i) {
```

Comment:
You may want to add comment here to avoid using `for (ConsumerAndMetadata consumerAndMetadata : consumerAndMetadatas)`

Reply:
Done